src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/ReadAndUpdateOperation.java

```diff
@@ -26,47 +28,94 @@ public record ReadAndUpdateOperation(
     boolean returnDocumentInResponse,
     boolean returnUpdatedDocument,
     boolean upsert,
-    Shredder shredder)
+    Shredder shredder,
+    int updateLimit)
     implements ModifyOperation {
 
   @Override
   public Uni<Supplier<CommandResult>> execute(QueryExecutor queryExecutor) {
-    Uni<ReadOperation.FindResponse> docsToUpate = readOperation().getDocuments(queryExecutor, null);
+    final AtomicBoolean moreDataFlag = new AtomicBoolean(false);
+    final Multi<ReadOperation.FindResponse> findResponses =
+        Multi.createBy()
+            .repeating()
+            .uni(
+                () -> new AtomicReference<String>(null),
+                stateRef -> {
+                  Uni<ReadOperation.FindResponse> docsToDelete =
```

Comment:
Should it be named `docsToDelete` if this is used for other operations (update) too?
Unless this is only for deletions.

Reply:
This class is for update, will rename to docsToUpdate. My bad, copy & paste mistake.

Reply:
np.